server/src/test/java/org/elasticsearch/transport/RemoteClusterCredentialsManagerTests.java

```diff
@@ -20,18 +22,80 @@ public void testResolveRemoteClusterCredentials() {
         final String clusterAlias = randomAlphaOfLength(9);
         final String otherClusterAlias = randomAlphaOfLength(10);
 
```

Comment:
If the `CredentialsManager` is expected to be thread safe, a concurrency test could be useful in this class.

Reply:
I'll log a follow up item to address this one in Jira to not block getting this out the door.